### PR TITLE
Repo metadata load and save

### LINKED ISSUE #54: Feature Request: Have programatic way of adding metadata to a repo
If a user creates dozens of repos programmatically, and then realized they forgot adding some tag or any other info for the metadata, they will have to either go through each of them manually, re-generate the README.md, or change the README.md content manually which can be error-prone.

### PR DESCRIPTION
**Pull request by julien-c**
Tentatively close #54 

Example use case:
- goal: updating all licenses for Helsinki-NLP models with no license, to `apache-2.0` – see https://github.com/huggingface/transformers/issues/13328#issuecomment-916455471
- example script: https://gist.github.com/julien-c/b2dcde5df5d5e41ad7c4b594cb54aba3

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
That's definitely useful, thanks a lot for adding this!

**Comment by osanseviero** [comment]
Thanks so much for this! This is a very needed upgrade! ❤️‍🔥 

This will be very useful for all our 3rd party integrations if we ever need to modify all the repos. We should communicate about this PR once merged to all big libraries and orgs (e.g. Stanford NLP)

**Comment by julien-c** [comment]
(current status: spamming the CI to try and get all transient errors to go green 😅)

File: src/huggingface_hub/repocard.py
Comment (osanseviero) [inline src/huggingface_hub/repocard.py:9]: nit:

```suggestion
# exact same regex as in the Hub server. Please keep in sync.
REGEX_YAML_BLOCK = re.compile(r"---[\n\r]+([\S\s]*?)[\n\r]+---[\n\r]")
```

Also it would be good to add similar docstring in the Hub code

**Review by osanseviero — APPROVED** [review]
Good stuff!

File: src/huggingface_hub/constants.py
Comment (LysandreJik) [inline src/huggingface_hub/constants.py:16]: You're setting `MODELCARD_NAME` here but using the `repocard` term elsewhere - what should it be? I would favor having all instances be `REPOCARD_NAME` if you're looking for a repo-type-agnostic name

File: src/huggingface_hub/repocard.py
Comment (LysandreJik) [inline src/huggingface_hub/repocard.py:23]: This should probably raise an error too, right? Running `metadata_load` on any random file will return nothing while I would expect it to fail if it doesn't find any metadata

Comment (LysandreJik) [reply src/huggingface_hub/repocard.py:23]: If that's the expected behavior then I would also put it in a test to ensure that it doesn't switch to raising an error in the future

**Review by LysandreJik — APPROVED** [review]
Thank you for working on that @julien-c, that's very useful!

As seen with @elishowk this morning it would be nice to eventually have the same validation here as what is enforced in the backend. `transformers` could then leverage it.

File: src/huggingface_hub/repocard.py
Comment (julien-c) [reply src/huggingface_hub/repocard.py:23]: as discussed offline we have a lot of repo cards (model cards, etc) that do not have a yaml block, and i think the expected behavior here is to return None. If need be, we might modify this in a subsequent PR

**Comment by julien-c** [comment]
@LysandreJik attempted to solve merge conflict

This should be ready to merge, depending on final maintainer acceptance (😘 )

**Review by elishowk — APPROVED** [review]
LGTM

**Comment by LysandreJik** [comment]
Will push an additional test and merge in a bit.

**Comment by LysandreJik** [comment]
Thank you for your work @julien-c

**Comment by elishowk** [comment]
> (current status: spamming the CI to try and get all transient errors to go green sweat_smile)

Note about that issue I witnessed working on PRs : huggingface_hub and transformers CI often fails because of moon-staging 504 timeouts. If needed, I can work on strengthening this staging server's capacities

**Pull request merged**
